File: backend/src/main/kotlin/ru/disenchanted/backend/domain/activity/ChildActivity.kt
package ru.disenchanted.backend.domain.activity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ChildActivity(
    @SerialName("_id")
    val id: String,

    @SerialName("name")
    val name: String,

    @SerialName("logoUrl")
    val logoUrl: String?,

    @SerialName("parent")
    val parent: Activity?
)

File: web/src/lib/activity/ChildActivity.ts
import { Activity } from "$lib/activity/Activity"
import { Expose, Type } from "class-transformer"

export class ChildActivity {
    @Expose({ name: "_id" })
    readonly id: string

    readonly name: string

    readonly logoUrl: string

    @Type(() => Activity)
    readonly parent: Activity | null
}
